src/datasets.c

```diff
@@ -1372,7 +1373,12 @@ static int DatasetAddString(Dataset *set, const uint8_t *data, const uint32_t da
     struct THashDataGetResult res = THashGetFromHash(set->hash, &lookup);
     if (res.data) {
         DatasetUnlockData(res.data);
-        return res.is_new ? 1 : 0;
+        if (res.is_new) {
+            THashIncrMemuse(set->hash, data_len);
```

Comment:
why is this needed? can we not use the getter inside the THashGetFromHash logic?

Reply:
Indeed we can and that's the better way. I did not try to make the existing code from the last PR better. My bad.